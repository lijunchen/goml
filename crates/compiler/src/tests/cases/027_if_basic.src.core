fn choose(flag/0: bool, x/1: int, y/2: int) -> int {
  if flag/0 {
      x/1
  } else {
      y/2
  }
}

fn main() -> unit {
  let yes/3 = choose(true, 10, 99) in
  let no/4 = choose(false, 10, 99) in
  let mtmp0 = string_println(string_add("yes=", int_to_string(yes/3))) in
  let mtmp1 = string_println(string_add("no=", int_to_string(no/4))) in
  ()
}