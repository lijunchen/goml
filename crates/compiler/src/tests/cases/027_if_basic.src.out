yes=10
no=99
